### PR TITLE
Convert Modal and remaining Button subcomponents to TypeScript

### PR DESCRIPTION
**Pull request by pvenable**

### CODE REVIEW

File: src/Button/Button.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ const Button: FC<Props> = forwardRef<HTMLButtonElement, Props>(
       iconEnd,
       iconStart,
       intent = 'brand',
-      paletteColor = '',
+      paletteColor,
```
Comment (pvenable) [inline src/Button/Button.tsx:35]: `''` isn't a valid `PaletteColor`, but `undefined` should result in the same behavior.

File: src/Button/sizeStyles.js [hunk 2]
```diff
@@ -1,56 +0,0 @@
-import { css } from '@emotion/react';
```
Comment (pvenable) [inline src/Button/sizeStyles.js:1]: This seems to have been orphaned as of #248.

File: src/Modal/__tests__/Modal.test.tsx [hunk 3]
```diff
@@ -32,15 +37,15 @@ describe('<Modal />', () => {
     const mockStyle = css`
       background: orange;
     `;
-    buildModalCss.mockReturnValue(mockStyle);
+    mocked(buildModalCss).mockReturnValue(mockStyle);
```
Comment (pvenable) [inline src/Modal/__tests__/Modal.test.tsx:40]: https://kulshekhar.github.io/ts-jest/docs/guides/test-helpers/#mockedtitem-t-deep--false

File: src/Modal/__tests__/Modal.test.tsx [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
 import { Modal } from '../Modal';
 
 jest.mock('../buildModalCss');
 
+const isOpen = false;
+
 describe('<Modal />', () => {
   it('renders a Global for the modal styles', () => {
-    const modal = shallow(<Modal>My Modal</Modal>);
+    const modal = shallow(<Modal {...{ isOpen, theme }}>My Modal</Modal>);
```
Comment (pvenable) [inline src/Modal/__tests__/Modal.test.tsx:16]: `isOpen` and `theme` are required props; this just wasn't sufficiently expressed via PropTypes.

Comment (pvenable) [reply src/Modal/__tests__/Modal.test.tsx:16]: Though note that `theme` isn't actually required for consumers per se as the component is publicly exported `withTheme`.

File: src/Modal/__tests__/Modal.test.tsx [hunk 5]
```diff
@@ -49,18 +54,22 @@ describe('<Modal />', () => {
...
-    const modal = shallow(<Modal myProp="foo">My Modal</Modal>);
+    const modal = shallow(
+      <Modal {...{ isOpen, theme }} className="foo">
+        My Modal
+      </Modal>
+    );
 
     expect(modal.find(ReactModal)).toHaveProp({
-      myProp: 'foo'
+      className: 'foo'
```
Comment (pvenable) [inline src/Modal/__tests__/Modal.test.tsx:64]: Switched to test with a real prop.

File: src/Modal/__tests__/Modal.test.tsx [hunk 6]
```diff
@@ -49,18 +54,22 @@ describe('<Modal />', () => {
...
     });
   });
 
   it('does not pass theme to the Reactmodal', () => {
-    const modal = shallow(<Modal theme="foo">My Modal</Modal>);
+    const modal = shallow(<Modal {...{ isOpen, theme }}>My Modal</Modal>);
 
     expect(modal.find(ReactModal)).not.toHaveProp({
-      theme: 'foo'
+      theme
```
Comment (pvenable) [inline src/Modal/__tests__/Modal.test.tsx:72]: A string is not a valid theme.

File: src/Modal/buildModalCss.ts [hunk 7]
```diff
@@ -21,7 +22,17 @@ const baseOverlayCss = (theme) => css`
   top: 0;
 `;
 
-const buildModalCss = ({ theme, overlayCss, modalCss }) => css`
+export type Css = CSSInterpolation | ((theme: Theme) => CSSInterpolation);
```
Comment (pvenable) [inline src/Modal/buildModalCss.ts:25]: `CSSInterpolation` allows the typical stuff like an object with CSS property key/values, or an emotion `css` construct.

**Review by xxx — APPROVED** [review]

**Pull request merged**